website/docs/r/compute_route.html.markdown

```diff
@@ -40,52 +71,94 @@ resource "google_compute_route" "default" {
 
 The following arguments are supported:
 
-* `name` - (Required) A unique name for the resource, required by GCE.
-    Changing this forces a new resource to be created.
-
-* `dest_range` - (Required) The destination IPv4 address range that this
-    route applies to.
+* `dest_range` -
+  (Required)
+  The destination range of outgoing packets that this route applies to.
+  Only IPv4 is supported.
+* `name` -
+  (Required)
+  Name of the resource. Provided by the client when the resource is
+  created. The name must be 1-63 characters long, and comply with
+  RFC1035.  Specifically, the name must be 1-63 characters long and
+  match the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?` which means
+  the first character must be a lowercase letter, and all following
+  characters must be a dash, lowercase letter, or digit, except the
+  last character, which cannot be a dash.
+* `network` -
+  (Required)
+  The network that this route applies to.
 
-* `network` - (Required) The name or self_link of the network to attach this route to.
-
-* `priority` - (Optional) The priority of this route, used to break ties. Defaults to 1000.
 
 - - -
 
-* `next_hop_gateway` - (Optional) The URL of the internet gateway to route
-    to if this route is matched. The alias "default-internet-gateway" can also
-    be used.
+* `description` -
+  (Optional)
+  An optional description of this resource. Provide this property
+  when you create the resource.
+* `priority` -
+  (Optional)
+  The priority of this route. Priority is used to break ties in cases
+  where there is more than one matching route of equal prefix length.
+
+  In the case of two routes with equal prefix length, the one with the
+  lowest-numbered priority value wins.
+
+  Default value is 1000. Valid range is 0 through 65535.
+* `tags` -
+  (Optional)
+  A list of instance tags to which this route applies.
+* `next_hop_gateway` -
+  (Optional)
+  URL to a gateway that should handle matching packets.
+
+  Currently, you can only specify the internet gateway, using a full or
+  partial valid URL:
```

Comment:
We also allow the string "default-internet-gateway". While I'm thinking about it, maybe that should be handled as a field helper thing? But also let's put it in the docs.

Reply:
This is a fairly custom thing already since the URL is of a non-existent type 'gateways'. But yeah, this is pretty much already what we do for resourcerefs.